clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3322,6 +3316,17 @@ private int processKafkaDataMessage(
           keyBytes = leaderProducedRecordContext.getKeyBytes();
           delete = ((Delete) leaderProducedRecordContext.getValueUnion());
         }
+
+        if (recordTransformer != null) {
+          SchemaEntry keySchema = schemaRepository.getKeySchema(storeName);
+          Lazy<Object> lazyKey = Lazy.of(() -> deserializeAvroObjectAndReturn(ByteBuffer.wrap(keyBytes), keySchema));
+          Object record = recordTransformer.processDelete(lazyKey);
+
+          if (record != null) {
+            break;
```

Comment:
Is this right?  Is there not a scenario here where we'd delete from the local storage engine?

Reply:
The delete method originally returned an object before I took over. Maybe there was some motivation that some clients wanted to keep some objects in storage? If not, I can go ahead and change the return type to void.

Reply:
After discussing verbally, if we want the user to proceed with deletion in DaVinci they should return null. If they are deleting in an external storage they should return an object.